internal/controllers/machineset/machineset_controller.go

```diff
@@ -313,6 +313,10 @@ func (r *Reconciler) reconcile(ctx context.Context, cluster *clusterv1.Cluster,
 
 	// Always updates status as machines come up or die.
```

Comment:
Please rebase onto latest main. It's really misleading in the diff if surrounding code is outdated (like the handling of ErrClusterLocked in the reconcile func)

Reply:
> BTW can we cherry-pick this patch to release-1.6 or 1.5 ? We're using CAPI 1.5 now and don't want to jump to 1.7 right now.

In my opinion, yes. I think it's fair to consider this a bugfix and then it qualifies for backport into all active releases (1.6 and 1.5 (until the 1.7 release in ~ 2 weeks))

Reply:
Oh didn't realize that the PR was opened against release-1.5 :). But we have to merge against main first anyway.